dali/pipeline/data/buffer.h

```diff
@@ -433,26 +510,31 @@ class DLL_PUBLIC Buffer {
       if (grow > new_num_bytes) new_num_bytes = grow;
       reserve(new_num_bytes);
     } else if (!is_pinned() && new_num_bytes < num_bytes_ * shrink_threshold_) {
-      data_.reset();
-      num_bytes_ = 0;
+      free_storage();
       reserve(new_num_bytes);
     }
   }
 
  protected:
   void move_buffer(Buffer &&buffer) {
-    type_         = std::move(buffer.type_);
-    data_         = std::move(buffer.data_);
-    allocate_     = std::move(buffer.allocate_);
-    size_         = buffer.size_;
-    num_bytes_    = buffer.num_bytes_;
-    device_       = buffer.device_;
-    shares_data_  = buffer.shares_data_;
-    pinned_       = buffer.pinned_;
-
+    swap(buffer);
     buffer.reset();
   }
 
+  void free_storage(AccessOrder order = {}) {
+    if (data_) {
+      if (!order)
+        order = order_;
+      if (!set_deletion_order(data_, order))  // TODO(michalz): Check if it helps or hurts.
```

Comment:
When and how it is going to be determined?

Reply:
When: when we implement some recycling within the pipeline. For now, it seems that it works fine.